configure.ac

```diff
@@ -466,23 +475,36 @@ AC_ARG_ENABLE(network,
               AS_HELP_STRING([--disable-network],
                              [Disable networking features (modem, ipx)]),,
               enable_network=yes)
-AC_CHECK_HEADER(SDL_net.h,have_sdl_net_h=yes,)
+AC_CHECK_HEADER(SDL_net.h, have_sdl_net_h=yes,)
 AC_CHECK_LIB(SDL2_net, SDLNet_Init, have_sdl_net_lib=yes, , )
 AC_MSG_CHECKING([whether networking features will be enabled])
 if test x$enable_network = xyes ; then
   if test x$have_sdl_net_lib = xyes -a x$have_sdl_net_h = xyes ; then
-    LIBS="$LIBS -lSDL2_net"
+    libtype=""
+    if test x$enable_sdl_static = xyes ; then
+      libtype="(static)"
+      case "$host" in
+        *-*-darwin*)
+          LIBS="$LIBS /usr/local/lib/libSDL2_net.a"
+          ;;
+        *)
+          LIBS="$LIBS -Wl,-Bstatic -lSDL2_net -Wl,-Bdynamic"
```

Comment:
Suggestion: before line 491:

    AC_MSG_WARN([Static linking of SDL2 does not work reliably. Make sure you have libSDL2_net.a file available.])

Reply:
Added.

For PNG, I was able to replace the old style handling with PKG_CONFIG (and specifying a minimum version, which is nice); but unfortunately this trick didn't work for SDL2_Net because it's output include -lSDL2 (in addition to -lSDL2_net), and wrapping this statically: `libsdl2net_LIBS="-Wl,-Bstatic $libsdl2net_LIBS -Wl,-Bdynamic"` causes subsequent `./configure` link checks to fail - so it's a no go.

That said, after `./configure` is done, you can drop `-Wl,-Bstatic $libsdl2net_LIBS -Wl,-Bdynamic` into the Makefile, and it works no problem and produces a fully-static dosbox bin. Argh.  Autotools is a nightmare.